generator/src/Data/Shortcut/Modifiers.purs

```diff
@@ -0,0 +1,96 @@
+{-
+Copyright 2015 SlamData, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-}
+
+module Data.Shortcut.Modifiers where
+
+import Prelude
+import Data.Char (fromCharCode)
+import Data.KeyboardEvent (KeyboardEvent())
+import Data.Shortcut.Platform (Platform(..))
+import Data.String (joinWith, fromChar)
+
+data KeyModifiers = KeyNone | KeyMod
+
+data CodeModifiers = None | Alt | Shift | Mod | AltShift | AltMod | ShiftMod | AltShiftMod
```

Comment:
I agree and initially I supported this as well as `⌃⌘X`. I decided to not allow for `Ctrl` to be added directly as the `Windows` key doesn't register as a modifier key `event.metaKey` on windows. This means that if you define a shortcut such as `ctrlMetaX` it will only work on Apple platforms. This left me with 

`data CodeModifiers = None | Alt | Shift | Ctrl | Mod | AltShift | AltCtrl | AltMod | AltShiftCtrl | ShiftMod | AltShiftMod`

But on non apple platforms `Ctrl` and `Mod` are equivalent and I thought this was confusing especially as people who are used to non apple platforms may instinctively use `Ctrl` instead of `Mod` providing a less familiar experience for Apple users. @jdegoes said to me "Each platform should have as close to native experience as we can manage." so to make this library promote this idea I removed the `Ctrl` constructor from `CodeModifiers`.

Its also worth noting that `CodeModifiers` only work with KeyboardEvent `code` values which only give US layout results so they would not be used for shortcuts containing `X`.